bazel_tools/proto.bzl

```diff
@@ -286,13 +289,13 @@ def proto_jars(
     # JAR containing the generated Scala bindings.
     proto_gen(
         name = "%s_scala_sources" % name,
-        srcs = _proto_scala_srcs(name, grpc),
+        srcs = _proto_scala_srcs(name, grpc, java_conversions),
         plugin_exec = "//scala-protoc-plugins/scalapb:protoc-gen-scalapb",
         plugin_name = "scalapb",
-        plugin_options = ["grpc"] if grpc else [],
+        plugin_options = (["grpc"] if grpc else []) + (["java_conversions"] if java_conversions else []),
```

Comment:
Fix 1: making sure that the `proto_gen` rule is aware that the `java_conversions` flag needs to be passed.